Adjustable Goliath Cloaks

Dawnseer opened the pull request:
## About The Pull Request
I've enabled goliath cloaks to be alt clicked to adjust it to neck slot instead of over clothing slot.  
While adjusted to neck slot, it has no armor values for balance purposes, while in over clothing slot, it has the same armor values as usual. The hood has the same values as usual, as it still inhabits the same slot as usual.

![image](https://user-images.githubusercontent.com/126404225/231837773-97f79c0e-9de2-4bfa-bfec-ed6c67e0d476.png)
![image](https://user-images.githubusercontent.com/126404225/231837688-f1f6ea67-8127-423e-96a0-b125f7778194.png)
![image](https://user-images.githubusercontent.com/126404225/231837716-805f778b-4db1-4dd5-a667-c196524f023f.png)
## Why It's Good For The Game
It enables a cloak to function like a cloak. It gives people the option to use it not for armor, but for fashion.
## Changelog
:cl:
add: Goliath cloak can be worn as a cloak that doesn't provide any armor benefits
/:cl:

Dawnseer committed:
AdjustableGoliathCloaks

Makes goliath cloaks adjustable

Dawnseer commented:
I really do gotta learn github a bit better.

Dawnseer committed:
Update explorer_gear.dm

Sneaky bugger

MrDoomBringer commented on code/modules/mining/equipment/explorer_gear.dm:
```diff
@@ -130,6 +130,28 @@
+		if(slot_flags & ITEM_SLOT_OCLOTHING)
+			slot_flags = ITEM_SLOT_NECK
+			set_armor(/datum/armor/none)
+			to_chat(user,span_notice("You adjust [src] to fit around your neck."))
```
Make these use the visible message popup thing
Also could probably mention that it removes armor on neck

Dawnseer replied:
Alright, gave it visible message popup, and made you and others able to discern if it's for neck(unarmored) or overclothing(armored) via 'ceremonial use' and 'defensive use'
Also cleaned the code a bit up to make it usable for more things by using [src] instead of 'goliath cloak'

Dawnseer committed:
Visible adjustments, discerning use

Made the requested changes, and also cleaned the code a bit with more [src] instead of 'goliath cloak' so it can be used elsewhere.

Dawnseer committed:
Muh patterns

Changed a 'the' to 'your'

Dawnseer committed:
Merge branch 'master' into AdjustableGoliathCloakV2

Time-Green commented on code/modules/mining/equipment/explorer_gear.dm:
```diff
@@ -130,6 +130,28 @@
+/obj/item/clothing/suit/hooded/cloak/goliath/AltClick(mob/user)
+	. = ..()
+	if(iscarbon(user))
+		var/mob/living/carbon/C = user
```
Don't use single-letter vars please

Time-Green commented on code/modules/mining/equipment/explorer_gear.dm:
```diff
@@ -130,6 +130,28 @@
+		if((C.get_item_by_slot(ITEM_SLOT_NECK) == src))
+			to_chat(user, span_warning("You can't adjust [src] while wearing it!"))
+			return
+		if(C.get_item_by_slot(ITEM_SLOT_OCLOTHING == src))
```
An OR could work for this, but if you don't want to, can you use an 'else if' here? No point checking both of these since you wont be wearing it in your neck and on your exosuit slot at once

Dawnseer replied:
Alright I have no idea how to fit in an 'OR' I could not find any function from that, but || seemed to do what I imagine OR would.

Time-Green replied:
|| is OR, I just find it easier to say in conversation

Dawnseer replied:
Oh.

Time-Green commented on code/modules/mining/equipment/explorer_gear.dm:
```diff
@@ -130,6 +130,28 @@
+		if(slot_flags & ITEM_SLOT_OCLOTHING)
+			slot_flags = ITEM_SLOT_NECK
+			set_armor(/datum/armor/none)
+			user.visible_message(span_notice("[user] adjusts their [src] for ceremonial use."),span_notice("You adjust your [src] for ceremonial use"))
```
Can you use a space after your comma's here?

Time-Green commented on code/modules/mining/equipment/explorer_gear.dm:
```diff
@@ -130,6 +130,28 @@
+		else
+			slot_flags = initial(slot_flags)			
+			set_armor(initial(armor_type))
+			user.visible_message(span_notice("[user] adjusts their [src] for defensive use"),span_notice("You adjust your [src] for defensive use"))
```
Ditto

Dawnseer committed:
cleaned up the code

better spacing, less redundancy

Dawnseer committed:
Merge branch 'AdjustableGoliathCloakV2' of https://github.com/Dawnseer/tgstation into AdjustableGoliathCloakV2

Dawnseer committed:
Merge branch 'master' into AdjustableGoliathCloakV2

Fikou commented on code/modules/mining/equipment/explorer_gear.dm:
```diff
@@ -130,6 +130,25 @@
+		if(slot_flags & ITEM_SLOT_OCLOTHING)
+			slot_flags = ITEM_SLOT_NECK
+			set_armor(/datum/armor/none)
+			user.visible_message(span_notice("[user] adjusts their [src] for ceremonial use."), span_notice("You adjust your [src] for ceremonial use"))
+		else
+			slot_flags = initial(slot_flags)			
+			set_armor(initial(armor_type))
+			user.visible_message(span_notice("[user] adjusts their [src] for defensive use"), span_notice("You adjust your [src] for defensive use"))
```
periods at the end

Dawnseer committed:
Periods at the end

Probably what he meant by this?
